File: week5/knight.py
steps = [(-2, -1), (-1, -2), (1, -2), (2, -1), (2, 1), (1, 2), (-1, 2), (-2, 1)]

knight = input()
row = int(knight[1])
column = int(ord(knight[0])) - int(ord('a')) + 1

result = 0

for step in steps:
    Row = row + step[0]
    Col = column + step[1]

    if Row >= 1 and Row <= 8 and Col >= 1 and Col <= 8:
        result += 1

print(result)

File: week5/knight.c
#include <stdio.h>

int dx[] = { -2, -2, 2, 2, -1, -1, 1, 1 };
int dy[] = { -1, 1, -1, 1, -2, 2, -2, 2 };
char knight[3];

void main() {
	scanf("%s", knight);
	int row = knight[1] - '0';
	int column = knight[0] - 'a' + 1;
	int result = 0;

	for (int i = 0; i < 8; i++) {
		int Row = row + dx[i];
		int Col = column + dy[i];

		if (Row >= 1 && Row <= 8 && Col >= 1 && Col <= 8)
			result++;
	}
	printf("%d\n", result);

}